src/metabase/models/field_usage.clj

```diff
@@ -0,0 +1,100 @@
+(ns metabase.models.field-usage
+  (:require
+   [metabase.lib.core :as lib]
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.hierarchy :as lib.hierarchy]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.util :as lib.util]
+   [metabase.lib.util.match :as lib.util.match]
+   [metabase.models.interface :as mi]
+   [metabase.query-processor.middleware.fetch-source-query :as fetch-source-query]
+   [metabase.util.malli :as mu]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]
+   [toucan2.tools.disallow :as t2.disallow]))
+
+(doto :model/FieldUsage
+  (derive :metabase/model)
+  (derive ::t2.disallow/update)
+  (derive :hook/created-at-timestamped?))
+
+(methodical/defmethod t2/table-name :model/FieldUsage [_model] :field_usage)
+
+(t2/deftransforms :model/FieldUsage
+  {:used_in                mi/transform-keyword
+   :aggregation_function   mi/transform-keyword
+   :breakout_temporal_unit mi/transform-keyword
+   :breakout_binning       mi/transform-json
+   :filter_op              mi/transform-keyword
+   :filter_args            mi/transform-json})
+
+(defn- filter->field-usage
+  [query stage filter-clause]
+  (let [filter-parts (lib/filter-parts query stage filter-clause)
+        field-id     (-> filter-parts :column :id)]
+    (when (int? field-id)
+      {:field_id    field-id
+       :used_in     :filter
+       :filter_op   (-> filter-parts :operator :short)
+       :filter_args (:args filter-parts)})))
```

Comment:
The fact the `filter_args` are a JSON array and we don't have one arg per row will mean we need to parse the queries in SQL, which means one query for each dialect. I suppose that's an okay price to pay because it makes storing this data easier (and we don't even have a use case for it), but I just want to make sure we're okay paying it in the future.

Reply:
as for future, I was hopeing we can use jsonb or sth like similar :)

Reply:
To be clear space and performance is not the concern I have here, so jsonb vs text doesn't make a difference. 

It's that any queries on this data will need implementing in all three dialects of the app DB if we want them to be fast. And this might not be trivial for H2. 

e.g. for postgres, I can imagine a query like this:

```
SELECT value, COUNT(*) AS count
FROM field_usage_test
CROSS JOIN UNNEST(string_to_array(values, ',')) AS t(value)
WHERE field_id = 100 AND filter_op = '='
GROUP BY value
ORDER BY count DESC;
```

We'll need to do some serious work to get this query working in H2. A guy seems to have achieved it [here](https://github.com/h2database/h2database/issues/1815#issuecomment-1363015627) by creating an alias for a function implemented in Java, but that's a lot of code.

Reply:
I wonder if we should just leave filter values out of this PR, and come back to it later when we actually need to use them. It's not clear how we're going to use them, and if we do use them, we'd probably want to have them stored differently than in a JSON array.

Reply:
removed filter_args in [ea4aff0](https://github.com/metabase/metabase/pull/39671/commits/ea4aff04d5d1a2d1c113db614a5b603841a0d45a)